### PR TITLE
fix: free draw elements move on flip

### PR DESCRIPTION
**Pull request by alex-kim-dev**
Single element, `shift + h` or `shift + v` or through the context menu:

Peek 2022-08-17 19-41

### CODE REVIEW

**Comment by alex-kim-dev** [comment]
It would be great to get a review for this pr too, thanks!

File: src/actions/actionFlip.ts [hunk 1]
```diff
@@ -8,8 +8,8 @@ import { AppState } from "../types";
 import { getTransformHandles } from "../element/transformHandles";
 import { isFreeDrawElement, isLinearElement } from "../element/typeChecks";
 import { updateBoundElements } from "../element/binding";
-import { LinearElementEditor } from "../element/linearElementEditor";
```
Comment (DanielJGeiger) [inline src/actions/actionFlip.ts:11]: Keep this import.

File: src/actions/actionFlip.ts [hunk 2]
```diff
@@ -146,13 +147,11 @@ const flipElement = (
...
   }
 
-  if (isLinearElement(element)) {
-    for (let index = 1; index < element.points.length; index++) {
-      LinearElementEditor.movePoints(element, [
-        { index, point: [-element.points[index][0], element.points[index][1]] },
-      ]);
-    }
-    LinearElementEditor.normalizePoints(element);
+  if (isLinearElement(element) || isFreeDrawElement(element)) {
```
Comment (DanielJGeiger) [inline src/actions/actionFlip.ts:150]:
```suggestion
  if (isFreeDrawElement(element)) {
```

File: src/actions/actionFlip.ts [hunk 3]
```diff
@@ -146,13 +147,11 @@ const flipElement = (
...
+    const points = rescalePoints(0, -width, element.points, true).map(
+      ([x, y]) => [x - finalOffsetX, y] as const,
+    );
+    mutateElement(element, { points });
   } else {
```
Comment (DanielJGeiger) [inline src/actions/actionFlip.ts:155]:
```suggestion
  } else if (isLinearElement(element)) {
    for (let index = 1; index < element.points.length; index++) {
      LinearElementEditor.movePoints(element, [
        { index, point: [-element.points[index][0], element.points[index][1]] },
      ]);
    }
    LinearElementEditor.normalizePoints(element);
  } else {
```

**Review by DanielJGeiger — CHANGES_REQUESTED** [review]
The code added by this PR seems good to me!  By removing the code for flipping linear elements, this PR caused a regression there.  My suggested changes essentially restore the removed code to fix the regression.

@dwelle @ad1992

**Comment by alex-kim-dev** [comment]
Closing because the issue is fixed already

**Pull request closed without merging**